javascripts/admin/pre-initializers/custom-field-types.js.es6

```diff
@@ -0,0 +1,82 @@
+import { withPluginApi } from "discourse/lib/plugin-api";
+import { on, observes } from "ember-addons/ember-computed-decorators";
+
+import { states as usaStates } from "../lib/data/usa";
+import { fieldTypesValidations } from "../lib/validations";
+import {
+  types,
+  getBuiltInFieldTypes,
+  PhoneFieldType,
+  StateFieldType,
+  ZipCodeFieldType
+} from "../lib/field-types";
+
+const initializeCustomFieldTypes = api => {
+  api.modifyClassStatic("model:user-field", {
+    fieldTypes() {
+      return getBuiltInFieldTypes().concat(
+        PhoneFieldType,
+        StateFieldType,
+        ZipCodeFieldType
+      );
+    }
+  });
+
+  api.modifyClass("component:user-field", {
+    classNameBindings: ["isValidFormat::error"],
+    isValidFormat: true,
+
+    @on("init")
+    enhanceFieldComponentValidation() {
+      // If there is no validation func for a custom userField we assume identity function
```

Comment:
Well... _identity_ is when we return the same argument. Here is just a function return true.